src/main/java/graphql/nadel/result/ResultComplexityAggregator.java

```diff
@@ -0,0 +1,46 @@
+package graphql.nadel.result;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class ResultComplexityAggregator {
+    private Integer totalNodeCount;
+    private LinkedHashMap<String, Integer> serviceNodeCountsMap;
+
+
+    public ResultComplexityAggregator(Integer totalNodeCount, LinkedHashMap<String, Integer> serviceNodeCountsMap) {
+        this.totalNodeCount = totalNodeCount;
+        this.serviceNodeCountsMap = serviceNodeCountsMap;
+    }
+
+    public Integer getTotalNodeCount() {
+        return totalNodeCount;
+    }
+
+    public LinkedHashMap<String, Integer> getServiceNodeCountsMap() {
+        return serviceNodeCountsMap;
+    }
+
+    public void addAndSetServiceNodeCount(String serviceFieldName, Integer nodeCount) {
+        serviceNodeCountsMap.putIfAbsent(serviceFieldName, 0);
+        serviceNodeCountsMap.put(serviceFieldName, serviceNodeCountsMap.get(serviceFieldName) + nodeCount);
+        totalNodeCount += nodeCount;
+    }
+
+    public Map<String, Object> snapshotResultComplexityData() {
+
+        Map<String, Object> resultComplexityMap = new LinkedHashMap<>();
+        resultComplexityMap.put("totalNodeCount", totalNodeCount);
+        resultComplexityMap.put("serviceNodeCounts", serviceNodeCountsMap);
+
+        return resultComplexityMap;
+    }
+
+    @Override
+    public String toString() {
+        return "ResultComplexityAggregator{" +
+                "totalNodeCount=" + totalNodeCount +
+                ", serviceNodeCountsMap=" + serviceNodeCountsMap +
+                '}';
+    }
+}
```

Comment:
There is no test for `ResultComplexityAggregator` that I can see

You should have a small unit test that it adds up per service just by itself.

I know the other tests test this - but you want a unit test per class (if you can) and then other tests to show that component interacts with other classes (but not to the same depth as the specific class unit test)